plugin/manager.go

```diff
@@ -86,11 +91,11 @@ type controller struct {
 // pluginRegistryService ensures that all resolved repositories
 // are of the plugin class.
 type pluginRegistryService struct {
-	registry.Service
+	EndpointResolver
 }
 
 func (s pluginRegistryService) ResolveRepository(name reference.Named) (repoInfo *registry.RepositoryInfo, err error) {
-	repoInfo, err = s.Service.ResolveRepository(name)
+	repoInfo, err = s.ResolveRepository(name)
```

Comment:
This is recursive? Also I don't think anything on `pluginRegistryService` implements `ResolveRepository` except itself.

Reply:
Yes, I was coming to the same conclusion; just pushed a commit that removes the whole `pluginRegistryService` type